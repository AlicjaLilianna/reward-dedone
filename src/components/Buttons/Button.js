import React, { useContext } from "react";
import styled from "@emotion/styled";
import { DrawerContext } from "../../providers/DrawerContext";

function Button(props) {
  //styles
  const Button = styled("button")`
    font-family: Poppins;
    border-radius: 15px;
    padding: 13px 20px;
    border: 1px solid transparent;
    font-size: 12px;
    &.btn-primary {
      min-width: 135px;
      background-color: #435f7c;
      color: white;
      &:active {
        background-color: darken(#435f7c, 10%);
      }
    }
    &.btn-add {
      min-width: auto !important;
      padding: 8px 16px;
      font-size: 18px;
      font-weight: 700;
      transform: rotate(5deg);
    }
    &.btn-secondary {
      background-color: transparent;
      min-width: 135px;
      border-color: #435f7c;
      color: #435f7c;
    }
  `;

  return (
    <Button
      role="button"
      className={props.btnClass}
      onClick={props.btnEvent}
      type={props.type}
      form={props.form}
    >
      {props.btnText}
    </Button>
  );
}

export default Button;
