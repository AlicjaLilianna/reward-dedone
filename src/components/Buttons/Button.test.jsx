import renderer from "react-test-renderer";
import Button from "./Button";

it("primary button renders correctly", () => {
  const tree = renderer.create(<Button buttonType="primary"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("secondary button renders correctly", () => {
  const tree = renderer
    .create(<Button buttonType="secondary"></Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("addIcon button renders correctly", () => {
  const tree = renderer.create(<Button buttonType="addIcon"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
