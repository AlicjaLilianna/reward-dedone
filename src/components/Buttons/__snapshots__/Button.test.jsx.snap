// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addIcon button renders correctly 1`] = `
<button
  className="button btnAdd btnPrimary"
  role="button"
/>
`;

exports[`primary button renders correctly 1`] = `
<button
  className="button btnPrimary"
  role="button"
/>
`;

exports[`secondary button renders correctly 1`] = `
<button
  className="button btnSecondary"
  role="button"
/>
`;
